stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -906,10 +907,119 @@ public class MessageListViewModel(
         val isUserMuted = chatClient.globalState.muted.value.any { it.target.id == user.id }
 
         if (isUserMuted) {
-            chatClient.unmuteUser(user.id)
+            unmuteUser(user.id)
         } else {
-            chatClient.muteUser(user.id)
-        }.enqueue()
+            muteUser(user.id)
+        }
+    }
+
+    /**
+     * Mutes the given user inside this channel.
+     *
+     * @param userId The ID of the user to be muted.
+     * @param timeout The period of time for which the user will
+     * be muted, expressed in minutes. A null value signifies that
+     * the user will be muted for an indefinite time.
+     */
+    public fun muteUser(
```

Comment:
The idea of exposing these as public is to allow the users to have an easier time of making up for lost functionality should they need it.